Makefile

```diff
@@ -149,7 +151,12 @@ endif
 	touch $@
 
 lint: .gopathok varlink_generate ## Execute the source code linter
+ifeq ($(PRE_COMMIT),)
+	$(warning pre-commit was not found, we will try to install it.)
+	$(PYTHON) -m pip install --user pre-commit
+endif
 	@echo "checking lint"
+	$(PRE_COMMIT) run -a
```

Comment:
It doesn't work that way.

Even if we allowed the auto-install, this wouldn't work because `PRE_COMMIT` would still be null.

This isn't critical, because I'm rejecting auto-install, but is just an FYI to encourage you to look at this sort of thing in the future.